pkg/util/manager/manager.go

```diff
@@ -0,0 +1,106 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
+)
+
+// typeIndexerFuncMap contains internal type and its indexerFunc used to set up indexer.
+var typeIndexerFuncMap = map[client.Object]client.IndexerFunc{
+	&workv1alpha2.ResourceBinding{}:        indexForResourceBinding(),
+	&workv1alpha1.ResourceBinding{}:        indexForResourceBinding(),
```

Comment:
I think there is no need to index `workv1alpha1.ResourceBinding`, `workv1alpha1.ClusterResourceBinding` either.